lib/Data/OpenSRS/ToXML.hs

```diff
@@ -14,122 +16,121 @@ import Text.XmlHtml
+                itemNode "f_lock_domain" $ boolVal requestLock,
+                itemNode "f_parkp" $ boolVal requestPark,
+                itemNode "f_whois_privacy" $ boolVal requestWhoisPrivacy,
+                itemNode "handle" $ boolVal' "process" "save" requestHandleNow,
+                itemNode "period" $ show requestPeriod,
+                itemNode "reg_username" $ show requestUsername,
+                itemNode "reg_password" $ show requestPassword,
+                itemNode "reg_type" $ show requestRegType
                 ])
-    tldData = case tld of
+    tldData = case requestTldData of
         Nothing -> []
         Just tld' -> [itemParent "tld_data" [tag "dt_assoc" $
             itemParentMap (\vmap -> [tag "dt_assoc" $ itemMap vmap]) tld'
             ]]
-requestXML (ChangeDomainPassword c domain_name password) = XmlDocument UTF8 doctype nodes
+requestXML ChangeDomainOwnership{..} = XmlDocument UTF8 doctype nodes
   where
-    nodes = wrapRequest $ genericRequest "CHANGE" "PASSWORD" (srsIpAddress c)
-        [("domain", domain_name),
-         ("reg_password", show password)]
-requestXML (SendDomainPassword c domain_name sendTo subUser) = XmlDocument UTF8 doctype nodes
+    nodes = wrapRequest $ genericRequest "CHANGE" "OWNERSHIP" (srsIpAddress requestConfig)
+        [("domain", requestDomainName),
+         ("username", show requestUsername),
+         ("password", show requestPassword)]
+requestXML SendDomainPassword{..} = XmlDocument UTF8 doctype nodes
   where
-    nodes = wrapRequest $ genericRequest "CHANGE" "PASSWORD" (srsIpAddress c)
-        [("domain_name", domain_name),
-         ("send_to", sendTo),
-         ("sub_user", boolVal subUser)]
-requestXML (SetCookie c d u p) = XmlDocument UTF8 doctype nodes
+    nodes = wrapRequest $ genericRequest "SEND_PASSWORD" "DOMAIN" (srsIpAddress requestConfig)
+        [("domain_name", requestDomainName),
+         ("send_to", requestSendTo),
+         ("sub_user", boolVal requestToSubuser)]
+requestXML SetCookie{..} = XmlDocument UTF8 doctype nodes
   where
-    nodes = wrapRequest $ genericRequest "SET" "COOKIE" (srsIpAddress c)
-        [("domain", d),
-         ("reg_username", show u),
-         ("reg_password", show p)]
+    nodes = wrapRequest $ genericRequest "SET" "COOKIE" (srsIpAddress requestConfig)
+        [("domain", requestDomainName),
+         ("reg_username", show requestUsername),
+         ("reg_password", show requestPassword)]
 
 -- | domain
 domainToNodes :: Domain -> [Node]
 domainToNodes domain = [
     itemNode "domain" (domainName domain),
     itemParent "contact_set" [
-        tag "dt_assoc" (concatMap contactToNodes $ toList $ domainContactSet domain)
+        tag "dt_assoc" ((=<<) contactToNodes . toList $ domainContactSet domain)
     ],
     itemParent "nameserver_list" [
-        tag "dt_array" (concatMap nsToNodes $ zip [0..] $ domainNameservers domain)
+        tag "dt_array" ((=<<) nsToNodes . zip [0..] $ domainNameservers domain)
```

Comment:
(=<<) is fine here, but, it's meant to be placed infix. Hlint is a bit dumb when it suggests to use it as (=<<)